webhooks-extension/pkg/endpoints/sink.go

```diff
@@ -245,6 +248,98 @@ func createPipelineRunFromWebhookData(buildInformation BuildInformation, r Resou
 	logging.Log.Debugf("PipelineRun created: %+v.", pipelineRun)
 }
 
+// This creates TaskRun for monitoring the main PipelineRun and reporting the result to the github
+func createTaskRunFromWebhookData(buildInformation BuildInformation, r Resource) {
+	logging.Log.Debugf("In createTaskRunFromWebhookData, build information: %s.", buildInformation)
+
+	installNs := os.Getenv("INSTALLED_NAMESPACE")
+	if installNs == "" {
+		installNs = "default"
+	}
+
+	logging.Log.Debugf("Looking for the pipeline configmap in the install namespace %s.", installNs)
+
+	// get information from related githubsource instance
+	webhook, err := r.getGitHubWebhook(buildInformation.REPOURL, installNs)
+	if err != nil {
+		logging.Log.Errorf("error getting github webhook: %s.", err.Error())
+		return
+	}
+
+	taskTemplateName := webhook.PullTask
+	taskNs := webhook.Namespace
+	saName := webhook.ServiceAccount
+	accessTokenRef := webhook.AccessTokenRef
+	OnSuccessComment := webhook.OnSuccessComment
+	OnFailureComment := webhook.OnFailureComment
+
+	if saName == "" {
+		saName = "default"
+	}
+
+	if taskTemplateName == "" {
+		taskTemplateName = "monitor-result-task"
+	}
+
+	if OnSuccessComment == "" {
+		OnSuccessComment = "OK"
+	}
+
+	if OnFailureComment == "" {
+		OnFailureComment = "ERROR"
+	}
+
+	logging.Log.Debugf("Build information: %+v.", buildInformation)
+
+	// Assumes you've already applied the yml: so the task definition must exist upfront.
```

Comment:
Given that the taskrun creation is always attempted ... shouldn't the task live in the config dir and the release yaml as it becomes a fundamental necessity of the install?

Reply:
I put the copy of the task in the config directory